compiler/rustc_next_trait_solver/src/solve/eval_ctxt/mod.rs

```diff
@@ -148,9 +148,9 @@ pub trait SolverDelegateEvalExt: SolverDelegate {
         goal: Goal<Self::Interner, <Self::Interner as Interner>::Predicate>,
         generate_proof_tree: GenerateProofTree,
         span: <Self::Interner as Interner>::Span,
-        cache_key: Option<CanonicalGoalCacheKey<Self::Interner>>,
+        stalled_on: Option<GoalStalledOn<Self::Interner>>,
     ) -> (
-        Result<(HasChanged, Certainty, CanonicalGoalCacheKey<Self::Interner>), NoSolution>,
+        Result<GoalEvaluation<Self::Interner>, NoSolution>,
```

Comment:
I gave the struct that corresponds to `(HasChanged, Certainty, CanonicalGoalCacheKey<I>)` the name `GoalEvaluation`, which coincidentally was also what we call it in `inspect::GoalEvaluation`. I think this parallel is actually good.